### PR TITLE
[MM-60075] Convert ./components/admin_console/team_channel_settings/group/group_members_modal.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by KvngMikey**
#### Summary
The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/27897
Jira https://mattermost.atlassian.net/browse/MM-60075

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]
Thank you so much! Just one more small minor thing from my side :smile:

File: webapp/channels/src/components/admin_console/team_channel_settings/group/group_members_modal.tsx [hunk 1]
```diff
@@ -15,78 +15,68 @@ type Props = {
...
-        };
-    }
-
-    componentDidMount() {
-        if (this.props.onLoad) {
-            this.props.onLoad();
+    useEffect(() => {
+        if (onLoad) {
+            onLoad();
         }
```
[the remark this answers is not in the document]
Comment (KvngMikey) [reply webapp/channels/src/components/admin_console/team_channel_settings/group/group_members_modal.tsx]: small, minor, tiny thing, just cause it's you, it has been addressed lol

**Review by Rajat-Dabade — APPROVED** [review]

File: webapp/channels/src/components/admin_console/team_channel_settings/group/group_members_modal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (KvngMikey) [reply webapp/channels/src/components/admin_console/team_channel_settings/group/group_members_modal.tsx]: so i should do this and update the PR right ?

**Comment by KvngMikey** [comment]
@larkox , can this be merged now ?

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**